aspnetcore-webapi/RemoteHostIpAddress/RemoteHostIpAddress/RemoteHostService.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.Net.Sockets;
+using System.Net;
+
+namespace RemoteHostIpAddress
+{
+    public class RemoteHostService : IRemoteHostService
+    {
+        public IPAddress GetRemoteHostIpAddressUsingRemoteIpAddress(HttpContext httpContext)
+        {
+            return httpContext.Connection.RemoteIpAddress;
+        }
+
+        public IPAddress? GetRemoteHostIpAddressUsingXForwardedFor(HttpContext httpContext)
+        {
+            IPAddress? remoteIpAddress = null;
+            var forwardedFor = httpContext.Request.Headers["X-Forwarded-For"].FirstOrDefault();
+
+            if (!string.IsNullOrEmpty(forwardedFor))
+            {
+                var ips = forwardedFor.Split(',', StringSplitOptions.RemoveEmptyEntries)
+                                      .Select(s => s.Trim());
+
+                foreach (var ip in ips)
+                {
+                    if (IPAddress.TryParse(ip, out var address) &&
+                        (address.AddressFamily == AddressFamily.InterNetwork || address.AddressFamily == AddressFamily.InterNetworkV6))
```

Comment:
this is taking a bit of space and can't be viewed in the article without scrolling, can we switch to using 
`address.AddressFamily is AddressFamily.InterNetwork or AddressFamily.InterNetworkV6` and if it still doesn't fit into the screen without screen maybe we can split into two lines.